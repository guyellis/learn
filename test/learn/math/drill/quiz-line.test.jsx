
const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const QuizLine = require('../../../../src/learn/math/drill/quiz-line');

const muiTheme = getMuiTheme(lightBaseTheme);

test('Badge totals should be rendered', () => {
  const component = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <QuizLine
        checkAnswer={jest.fn()}
        newRecord={{
          isNewRecord: false,
          currentTimePerQuestion: 1.1,
          existingRecordTimePerQuestion: 1,
        }}
        onscreenKeyboard={false}
        largeKeyboard={false}
        problem={[1, 1, 0, 2]}
      />
    </MuiThemeProvider>);
  const badgeTotals = component.toJSON();
  expect(badgeTotals).toMatchSnapshot();
});
