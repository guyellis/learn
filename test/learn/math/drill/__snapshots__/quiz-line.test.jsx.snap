// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge totals should be rendered 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <span
      style={
        Object {
          "fontSize": "xx-large",
          "margin": "10px",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "fontSize": "xx-large",
          "margin": "10px",
        }
      }
    >
      +
    </span>
    <span
      style={
        Object {
          "fontSize": "xx-large",
          "margin": "10px",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "fontSize": "xx-large",
          "margin": "10px",
        }
      }
    >
      =
    </span>
    <div
      className="MuiFormControl-root MuiTextField-root"
      onKeyPress={[Function]}
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={true}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="answer-number"
          name="answer"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value=""
        />
      </div>
    </div>
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiFab-root MuiFab-extended"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": "10px",
        }
      }
      tabIndex={0}
      title="Check Answer"
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "marginRight": "10",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        Check
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Ready for your first answer!
    </h5>
  </div>
  <div>
    <div
      style={
        Object {
          "border": "medium solid green",
          "borderRadius": "5px",
          "display": "inline-block",
          "paddingRight": "20px",
        }
      }
    >
      Current Speed 1.1 and Record 1
    </div>
  </div>
  <div />
</div>
`;
